File: client/node/tron.js
var io = require('socket.io-client');
var ai = require('./ai');

var socket = io('http://localhost:1337');

socket.on('connect', function () {
  socket.emit('join', ai.room, ai.team);
});

socket.on('start', function (config) {
  ai.start(config);
});

socket.on('next', function (prevMoves) {
  socket.emit('move', ai.next(prevMoves));
});

socket.on('view', function (grid) {
  console.log(grid);
});

socket.on('end', function (winnerID) {
  ai.end(winnerID);
  process.exit();
});

File: client/java/src/main/java/cybercybcles/Tron.java
package cybercybcles;

import io.socket.client.IO;
import io.socket.client.Socket;
import io.socket.emitter.Emitter;
import java.net.URISyntaxException;

public class Tron {
  public static void main(String[] args) throws URISyntaxException {
    Socket socket = IO.socket("http://localhost:1337");

    socket.on(Socket.EVENT_CONNECT, new Emitter.Listener() {
      @Override
      public void call(Object... args) {
        socket.emit("join", AI.ROOM, AI.TEAM);
      }
    }).on("start", new Emitter.Listener() {
      @Override
      public void call(Object[] config) {
        AI.start(config);
      }
    }).on("next", new Emitter.Listener() {
      @Override
      public void call(Object[] prevMoves) {
        socket.emit("move", AI.next(prevMoves));
      }
    }).on("view", new Emitter.Listener() {
      @Override
      public void call(Object[] grid) {
        System.out.println(grid);
      }
    }).on("end", new Emitter.Listener() {
      @Override
      public void call(Object[] winnerID) {
        AI.end(winnerID);
        System.exit(0);
      }
    });

    socket.connect();
  }
}
